hal/platform/atmelsam/irq/default_irq_handlers.c

```diff
@@ -47,27 +47,43 @@ volatile bustransaction_t sercom_bustrans_buffer[6] = {{SERCOMACT_NONE, 0, NULL,
 #define SERCOM_I2C_MASTER_RECV_ACK_AND_REQ_NEW_BYTE SERCOM_I2CM_CTRLB_CMD(2) | SERCOM_I2CM_CTRLB_SMEN
 #define SERCOM_I2C_MASTER_NACK_AND_STOP             SERCOM_I2CM_CTRLB_CMD(3) | SERCOM_I2CM_CTRLB_ACKACT | SERCOM_I2CM_CTRLB_SMEN
 
+
+void updateBusTransactionStates(Sercom* sercom_instance, volatile bustransaction_t* transaction) {
+  transaction->rx_status      = sercom_instance->I2CM.STATUS.bit.RXNACK;
+  transaction->bus_state      = sercom_instance->I2CM.STATUS.bit.BUSSTATE;
+  transaction->state_lenerr   = sercom_instance->I2CM.STATUS.bit.LENERR;
+  transaction->state_sexttout = sercom_instance->I2CM.STATUS.bit.SEXTTOUT;
+  transaction->state_mexttout = sercom_instance->I2CM.STATUS.bit.MEXTTOUT;
+  transaction->state_clkhold  = sercom_instance->I2CM.STATUS.bit.CLKHOLD;
+  transaction->state_lowtout  = sercom_instance->I2CM.STATUS.bit.LOWTOUT;
+  transaction->bus_arblost    = sercom_instance->I2CM.STATUS.bit.ARBLOST;
+  transaction->bus_error      = sercom_instance->I2CM.STATUS.bit.BUSERR;
+}
+
+
 /**
  * @brief Default IRQ Handler for the I2C master data send interrupt
  * @param hw Pointer to the HW peripheral to be manipulated
  * @param transaction The current transaction information
  */
 void i2c_host_data_send_irq(const void* hw, volatile bustransaction_t* transaction) {
-    Sercom*    sercom_instance = ((Sercom*)hw);
-    const bool write_buffer_exists = (transaction->write_buffer != NULL);
-    const bool has_bytes_left_to_write = (transaction->buf_cnt < transaction->buf_size);
-    if (write_buffer_exists && has_bytes_left_to_write) {
-        sercom_instance->I2CM.DATA.reg = transaction->write_buffer[transaction->buf_cnt++];
-    } else {
-        const bool send_stop_bit = (transaction->transaction_type != SERCOMACT_I2C_DATA_TRANSMIT_NO_STOP);
-        if (send_stop_bit) {
-            sercom_instance->I2CM.CTRLB.reg = SERCOM_I2C_MASTER_NACK_AND_STOP;
-        } else {
-            sercom_instance->I2CM.INTFLAG.reg = SERCOM_I2CM_INTFLAG_MB;
-        }
-        transaction->transaction_type = SERCOMACT_IDLE_I2CM;
-        transaction->buf_cnt = 0;
-    }
+  Sercom*    sercom_instance = ((Sercom*)hw);
+  const bool write_buffer_exists = (transaction->write_buffer != NULL);
+  const bool has_bytes_left_to_write = (transaction->buf_cnt < transaction->buf_size);
+  if (write_buffer_exists && has_bytes_left_to_write) {
+      sercom_instance->I2CM.DATA.reg = transaction->write_buffer[transaction->buf_cnt++];
+  } else {
+      const bool send_stop_bit = (transaction->transaction_type != SERCOMACT_I2C_DATA_TRANSMIT_NO_STOP);
+      if (send_stop_bit) {
+          sercom_instance->I2CM.CTRLB.reg = SERCOM_I2C_MASTER_NACK_AND_STOP;
+      } else {
+          sercom_instance->I2CM.INTFLAG.reg = SERCOM_I2CM_INTFLAG_MB;
+      }
+      transaction->transaction_type = SERCOMACT_IDLE_I2CM;
+      transaction->buf_cnt = 0;
+  }
+  updateBusTransactionStates(sercom_instance, transaction);
```

Comment:
Normally, I would let this slip but since I want to use this code more than just today. It might be better to refactor this to match the surrounding code style (**Uhm.. uhm.. Snake case**).

Also making this routine inline static void might benefit in this case (use cases for inline are rare). Since this subroutine contains such simple code that it might benefit from being inlined.